src/node.cc

```diff
@@ -3687,6 +3683,15 @@ void Init(int* argc,
     }
   }
 
+#ifdef __POSIX__
```

Comment:
Probably it will always collect samples during idle time (I have not tested Windows).
Note that the original code above was POSIX-only as well. It is moved here so that it can be conditional on the command line flag '--prof' (so only block SIGPROF if using v8.log-style profiling).